File: ios/Packages/MailClockKit/Sources/MailClockKit/PostalDays.swift
public func isPostalDay(_ isoDate: String) throws -> Bool {
    let date = try PostalDate(isoDate)
    if date.weekday == 1 {
        return false
    }
    return try !isObservedHoliday(isoDate)
}

public func postalDayOnOrAfter(_ isoDate: String) throws -> String {
    var date = try PostalDate(isoDate)
    for _ in 0..<30 {
        if try isPostalDay(date.description) {
            return date.description
        }
        date = try date.adding(days: 1)
    }
    throw MailClockError.noDeliveryDay(isoDate)
}

public func nextPostalDay(_ isoDate: String) throws -> String {
    let date = try PostalDate(isoDate)
    return try postalDayOnOrAfter(date.adding(days: 1).description)
}

public func addPostalDays(_ isoDate: String, count: Int) throws -> String {
    guard count >= 0 else {
        throw MailClockError.negativePostalDays(count)
    }
    var cursor = try PostalDate(isoDate).description
    for _ in 0..<count {
        cursor = try nextPostalDay(cursor)
    }
    return cursor
}

File: packages/mailclock/src/postalDays.ts
import { DateTime } from "luxon";
import { isObservedHoliday } from "./holidays.js";

/**
 * A postal day is a day the mail moves: Monday through Saturday, excluding
 * observed federal holidays. Sunday is never a postal day.
 */
export function isPostalDay(isoDate: string): boolean {
  const d = DateTime.fromISO(isoDate, { zone: "utc" });
  if (!d.isValid) throw new RangeError(`invalid date: ${isoDate}`);
  if (d.weekday === 7) return false;
  return !isObservedHoliday(isoDate);
}

/** The given date if it is a postal day, otherwise the next one. */
export function postalDayOnOrAfter(isoDate: string): string {
  let d = DateTime.fromISO(isoDate, { zone: "utc" });
  for (let i = 0; i < 30; i++) {
    const iso = d.toISODate()!;
    if (isPostalDay(iso)) return iso;
    d = d.plus({ days: 1 });
  }
  throw new Error(`no postal day within 30 days of ${isoDate}`);
}

/** The first postal day strictly after the given date. */
export function nextPostalDay(isoDate: string): string {
  return postalDayOnOrAfter(DateTime.fromISO(isoDate, { zone: "utc" }).plus({ days: 1 }).toISODate()!);
}

/** Advance `count` postal days from a date, not counting the start date itself. */
export function addPostalDays(isoDate: string, count: number): string {
  if (count < 0) throw new RangeError(`cannot advance a negative number of postal days: ${count}`);
  let cursor = isoDate;
  for (let i = 0; i < count; i++) cursor = nextPostalDay(cursor);
  return cursor;
}
